### ISSUE TITLE
`<charconv>`: `from_chars()` for `float` uses unnecessarily large structures

### ISSUE THREAD

**Issue by jrahlf** [microsoft/STL#2162]
I noticed that the `from_chars` implementations for both float and double use the `_Floating_point_string` structure, which is huge (almost 800byte). The structure appears to be allocated on the stack only.
I don't know how many bytes are needed for the float conversion, but certainly a lot less than the 800 bytes for the double conversion.

Maybe it is worth to use a (much) smaller structure for `from_chars` for float values?

**Comment by StephanTLavavej** [comment]
Yep, I was aware of this issue when I imported/refined the UCRT's `strtod()`/`strtof()` to make `from_chars()`. I should have documented this with a `// performance note` somewhere but didn't, so thanks for filing this issue - I have labeled it accordingly.

I agree with @AlexGuteniev that it would seem to have very minor benefits, which is why I didn't attempt to do this during the initial import. However, the necessary changes shouldn't be very extensive - the code is already templated on the floating-point type (as we must round from the input string to the floating-point type in a single step), so all that remains to be done is find all of the constants that need to be changed for `float` and provide the proper values. Test cases exercising the new boundary conditions would also be important to add - right now, I believe that the tests don't really bother with trying to inspect such cases for `float`, as I knew that the underlying code was using `double`-sized buffers.

**Comment by frederick-vs-ja** [comment]
In addition to `_Floating_point_string `, `_Big_integer_flt` can also be smaller for `float`.

https://github.com/microsoft/STL/blob/a62109595b6d89e08172fdf4beb75a2670fe0cc9/stl/inc/charconv#L424-L433

For `float` (IEEE 754 binary32 format), 113 digits should be sufficient (posted [here](https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/)).
